platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/eventhandling/DefaultTransactionHandlerTests.java

```diff
@@ -167,8 +173,9 @@ void freezeHandling() throws InterruptedException {
                 1,
                 handlerOutput.reservedSignedState().get().getReservationCount(),
                 "state should be returned with a reservation");
-
-        assertEquals(1, tester.getSubmittedActions().size(), "the freeze status should have been submitted");
+        // In addition to the freeze action, the uptime tracker reports a self event coming to consensus in the round.
+        assertEquals(2, tester.getSubmittedActions().size(), "the freeze status should have been submitted");
```

Comment:
Minor: I may be unaware of the internal details, but the generation of an extra event seems like a significant change in the behavior.

Why does this extra event get generated with this fix?

Reply:
In the code change, the `UptimeTracker.handleRound()` is no longer taking `AddressBook` as a second argument.  This unit testing was indirectly passing a null `AddressBook` and `UptimeTracker.handleRound()` was exiting due to the AB being null.   I removed the AB from the method because there is a roster in the round and the additional `AB` seems superfluous.    In the previous implementation, the `UptimeTracker` was exiting because the unit test is not standing up a complete platform state with a non-null address book.

Reply:
Got it. Thanks for the clarification!